oem/ibm/libpldmresponder/oem_ibm_handler.cpp

```diff
@@ -1063,15 +1126,52 @@ int encodeEventMsg(uint8_t eventType, const std::vector<uint8_t>& eventDataVec,
 
     return rc;
 }
+
+void pldm::responder::oem_ibm_platform::Handler::setDimmStateSensor(
+    bool status, uint16_t entityInstance)
+{
+    pldm::pdr::EntityType entityType = PLDM_ENTITY_MEMORY_MODULE;
+    pldm::pdr::StateSetId stateSetId = PLDM_OEM_IBM_SBE_DUMP_UPDATE_STATE;
+    uint8_t tid = TERMINUS_ID;
+
+    auto pdrs = findStateSensorPDR(tid, entityType, stateSetId, pdrRepo);
+    for (auto& pdr : pdrs)
+    {
+        auto stateSensorPDR =
+            reinterpret_cast<pldm_state_sensor_pdr*>(pdr.data());
+        if (entityInstance == stateSensorPDR->entity_instance)
+        {
+            if (status == true)
+            {
+                dumpStatusMap[entityInstance] = DimmDumpState::SUCCESS;
+                sendStateSensorEvent(stateSensorPDR->sensor_id,
+                                     PLDM_STATE_SENSOR_STATE, 0,
+                                     uint8_t(DimmDumpState::SUCCESS),
+                                     uint8_t(DimmDumpState::UNAVAILABLE));
```

Comment:
was the previous state really unavailable ? what happens if the retry operation worked ?